src/md.cc

```diff
@@ -674,5 +674,146 @@ void MGmol<OrbitalsType>::md(OrbitalsType** orbitals, Ions& ions)
+
+    /* For now, we only consider double-precision hdf5 I/O. */
+    assert(ct.restart_info > 3);
+    assert((ct.AtomsDynamic() == AtomsDynamicType::MD) || (ct.AtomsDynamic() == AtomsDynamicType::Quench));
+
+    HDFrestart h5file(filename, myPEenv, ct.out_restart_file_type);
+    int ierr;
+
+    OrbitalsType *restart_orbitals = nullptr;
+
+    /* This corresponds to MGmol<OrbitalsType>::initial */
+    {
+        // If not an initial run read data from files
+        if (ct.isLocMode())
+        {
+            std::string name = "ExtrapolatedFunction";
+            if (ct.verbose > 0)
+                printWithTimeStamp(
+                    "read LRs from ExtrapolatedFunction database...", os_);
+            int n = read_restart_lrs(h5file, name);
+            if (n == 0)
+            {
+                if (ct.verbose > 0)
+                    printWithTimeStamp("read LRs from Function database...", os_);
+                name = "Function";
+                n    = read_restart_lrs(h5file, name);
+            }
+            if (n < 0) return nullptr;
+
+            if (n > 0) lrs_->setup();
+        }
+
+        // Copy from current orbital, instead of constructing brand-new one
+        restart_orbitals = new OrbitalsType("ForLoading", *current_orbitals_, false);
+
+        // TODO(kevin): do we need this?
+        // increaseMemorySlotsForOrbitals<MemorySpaceType>();
+
+        // If not an initial run read data from files
+        md_time_      = h5file.getMDTimeFromFile();
+        md_iteration_ = h5file.getMDstepFromFile();
+
+        /* This corresponds to MGmol<OrbitalsType>::read_restart_data */
+        {
+            // TODO(kevin): do we need to load rho?
+            // read_rho_and_pot_hdf5(h5file, *rho_);
+
+            ierr = restart_orbitals->read_func_hdf5(h5file);
+        }   // read_restart_data
+
+        // TODO(kevin): Do we need this routine?
+        restart_orbitals->applyMask();
+
+    }   // initial()
+
+    /* This corresponds to MGmol<OrbitalsType>::md */
+    {
+        // TODO(kevin): do we need this?
+        // if (!ct.override_restart)
+        // {
+        //     if (onpe0) os_ << "Use restart file to initialize MD..." << std::endl;
+        //     stepper->init(h5file);
+        // }
+
+        int flag_extrapolated_data = 0;
+        if (onpe0)
+        {
+            flag_extrapolated_data
+                = h5file.dset_exists("ExtrapolatedFunction0000");
+            if (flag_extrapolated_data == 0)
+                flag_extrapolated_data
+                    = h5file.dset_exists("ExtrapolatedFunction0");
+        }
+        MPI_Bcast(&flag_extrapolated_data, 1, MPI_INT, 0, comm_);
+
+        /*
+            If extrapolated function exists,
+            then function is set as previous orbitals,
+            while the extrapolated function is set as the current orbitals.
+            This is how the restart file is saved via dumprestartFile.
+        */
+        if (flag_extrapolated_data)
```

Comment:
remove these 30 lines

Reply:
`setItCountLarge` and `moveVnuc` are unnecessary for now and commented.